erna/scripts/process_fact_mc.py

```diff
@@ -15,9 +15,32 @@
 
 logger = logging.getLogger(__name__)
 
+import re
+
+# given a string create a filename
+def create_filename_from_format(filename_format, basename, num):
+    """
+    Given a special format string, create a filename_format with the basename and a given number.
+    %b is replaced with the basename and is optional
+    %[0-9]?n is replaced with the number and must be supplied the number given will pad the given number with zeros
+    
+    Exp: create_filename_from_format("%b_%4n.fits", "test", 45)->"test_0045.fits"
```

Comment:
Why this complicated function using regex and stuff? Why not simply use pythons buildin? `{}_{:04d}`.format('test', 45)`?

Reply:
Why do you use the old and not guaranteed to stay `%` syntax and not the modern `'{}'`.format() syntax?.